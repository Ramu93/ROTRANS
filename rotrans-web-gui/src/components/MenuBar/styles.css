.menubar {
  width: 80px;
  height: 100%;
  background-color: #5026ac;
}

.menu {
  width: 80px;
  height: 80px;
  background-color: transparent;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.menu-no-pointer {
  cursor: default;
}

.menu-active {
  background-color: #6435c9;
}
