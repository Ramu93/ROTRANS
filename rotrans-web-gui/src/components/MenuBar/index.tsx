import React from "react";
import { Link, useLocation } from "react-router-dom";
import "./styles.css";

interface Props {
  port: string;
}

const MenuBar: React.FC<Props> = ({ port }) => {
  let activeMenu = 0;
  const location = useLocation();
  if (location.pathname.includes("visualization")) {
    activeMenu = 1;
  } else if (location.pathname.includes("transactions")) {
    activeMenu = 2;
  } else {
    activeMenu = 3;
  }

  return (
    <div className="menubar">
      <div className="menu menu-no-pointer">
        <img src={require("../../assets/svg/coin.svg")} />
      </div>
      <Link to={`/${port}/visualization`}>
        <div
          data-testid="vizMenuItem"
          className={activeMenu === 1 ? "menu menu-active" : "menu"}
        >
          <img src={require("../../assets/svg/board.svg")} />
        </div>
      </Link>
      <Link to={`/${port}/transactions`}>
        <div
          data-testid="transactionsMenuItem"
          className={activeMenu === 2 ? "menu menu-active" : "menu"}
        >
          <img src={require("../../assets/svg/rocket.svg")} />
        </div>
      </Link>
      <Link to={`/${port}/stake`}>
        <div
          data-testid="stakeMenuItem"
          className={activeMenu === 3 ? "menu menu-active" : "menu"}
        >
          <img
            src={require("../../assets/stake.png")}
            style={{ height: 30, width: 30 }}
          />
        </div>
      </Link>
    </div>
  );
};

export default MenuBar;
