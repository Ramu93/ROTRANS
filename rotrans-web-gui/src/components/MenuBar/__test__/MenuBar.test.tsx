import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import MenuBar from "..";

describe("MenuBar component", () => {
  afterEach(cleanup);

  it("Rendered visualization menu item", () => {
    const history = createMemoryHistory();
    const route = "/visualization";
    history.push(route);

    const { getByTestId } = render(
      <Router history={history}>
        <MenuBar />
      </Router>
    );
    expect(getByTestId("vizMenuItem")).toHaveClass("menu", "menu-active");
    expect(getByTestId("transactionsMenuItem")).toHaveClass("menu");
  });

  it("Rendered transactions menu item", () => {
    const history = createMemoryHistory();
    const route = "/transactions";
    history.push(route);

    const { getByTestId } = render(
      <Router history={history}>
        <MenuBar />
      </Router>
    );
    expect(getByTestId("vizMenuItem")).toHaveClass("menu");
    expect(getByTestId("transactionsMenuItem")).toHaveClass(
      "menu",
      "menu-active"
    );
  });

  it("Matches transactions snapshot ", () => {
    const history = createMemoryHistory();
    const route = "/transactions";
    history.push(route);

    const tree = renderer
      .create(
        <Router history={history}>
          <MenuBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches visualization snapshot ", () => {
    const history = createMemoryHistory();
    const route = "/visualization";
    history.push(route);

    const tree = renderer
      .create(
        <Router history={history}>
          <MenuBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
