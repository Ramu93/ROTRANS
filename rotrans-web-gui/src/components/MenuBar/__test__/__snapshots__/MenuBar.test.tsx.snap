// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar component Matches transactions snapshot  1`] = `
<div
  className="menubar"
>
  <div
    className="menu menu-no-pointer"
  >
    <img
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "coin.svg",
        }
      }
    />
  </div>
  <a
    href="/visualization"
    onClick={[Function]}
  >
    <div
      className="menu"
      data-testid="vizMenuItem"
    >
      <img
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "board.svg",
          }
        }
      />
    </div>
  </a>
  <a
    href="/transactions"
    onClick={[Function]}
  >
    <div
      className="menu menu-active"
      data-testid="transactionsMenuItem"
    >
      <img
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "rocket.svg",
          }
        }
      />
    </div>
  </a>
</div>
`;

exports[`MenuBar component Matches visualization snapshot  1`] = `
<div
  className="menubar"
>
  <div
    className="menu menu-no-pointer"
  >
    <img
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "coin.svg",
        }
      }
    />
  </div>
  <a
    href="/visualization"
    onClick={[Function]}
  >
    <div
      className="menu menu-active"
      data-testid="vizMenuItem"
    >
      <img
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "board.svg",
          }
        }
      />
    </div>
  </a>
  <a
    href="/transactions"
    onClick={[Function]}
  >
    <div
      className="menu"
      data-testid="transactionsMenuItem"
    >
      <img
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "rocket.svg",
          }
        }
      />
    </div>
  </a>
</div>
`;
